packages/vitest/src/node/pool.ts

```diff
@@ -391,14 +391,14 @@ function groupSpecs(specs: TestSpecification[], environments: Awaited<ReturnType
     }
 
     const order = spec.project.config.sequence.groupOrder
+    const isolate = spec.project.config.isolate
 
     // Files that have disabled parallelism and default groupOrder are set into their own group
-    if (order === 0 && spec.project.config.fileParallelism === false) {
+    if (isolate === true && order === 0 && spec.project.config.maxWorkers === 1) {
```

Comment:
The `isolate` check breaks cases where user want to run a single project sequentially by setting `fileParallelism: false`.

Reply:
Please, elaborate. There is a separate handling for `isolate: false` later on with a similar condition

Reply:
Yup that's right